micromagneticmodel/abstract/term.py

```diff
@@ -64,7 +64,9 @@ def __eq__(self, other):
         False
 
         """
-        if isinstance(other, self.__class__):
+        if (
+            isinstance(other, self.__class__) and self.name == other.name
+        ):  # To take into account user defined names
```

Comment:
[The following depends on whether we decide to use all attributes or not.]

Maybe this would be slightly easier:
```python
if isinstance(other, self.__class__):
    return self.name == other.name
```

Reply:
This creates a problem when the class of the terms are different. So we'll have to include an `else` statement anyways.

Reply:
The `else` case should stay. My suggestion was just to simplify/split the `if` case.

Reply:
Alternative:
```python
return isinstance(...) and self.name == other.name
```

Reply:
Good point!